controllers/agentmachine_controller.go

```diff
@@ -124,20 +126,20 @@ func (r *AgentMachineReconciler) Reconcile(ctx context.Context, req ctrl.Request
 	if agentMachine.Status.AgentRef == nil {
 		foundAgent, result, err := r.findAgent(ctx, log, agentMachine, agentCluster.Status.ClusterDeploymentRef, machineConfigPool, ignitionTokenSecretRef)
 		if err != nil || foundAgent == nil {
-			return result, err
+			return r.updateStatus(ctx, log, agentMachine, result, err, err == nil)
```

Comment:
I think this is clearer 
```suggestion
			return r.updateStatus(ctx, log, agentMachine, result, err, foundAgent == nil)
```